src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -262,6 +262,58 @@ FabricInfo * RetrieveCurrentFabric(CommandHandler * aCommandHandler)
     return Server::GetInstance().GetFabricTable().FindFabricWithIndex(index);
 }
 
+void FailSafeCleanup(FailSafeContext & failSafeContext)
+{
+    FabricInfo * fabricInfo = Server::GetInstance().GetFabricTable().FindFabricWithIndex(failSafeContext.GetFabricIndex());
+
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Call to FailSafeCleanup");
+
+    VerifyOrReturn(fabricInfo != nullptr);
+
+    // If an AddNOC or UpdateNOC command has been successfully invoked, terminate all CASE sessions associated with the Fabric
+    // whose Fabric Index is recorded in the Fail-Safe context (see ArmFailSafe Command) by clearing any associated Secure
+    // Session Context at the Server.
+    if (failSafeContext.NocCommandHasBeenInvoked())
+    {
+        CASESessionManager * caseSessionManager = Server::GetInstance().GetCASESessionManager();
+
+        if (caseSessionManager)
+        {
+            caseSessionManager->ReleaseSessionsForFabric(fabricInfo->GetCompressedId());
```

Comment:
Funny enough, this does not terminate all CASE sessions associated with the fabric.  In fact, it terminates none of them.

What should really be happening here is a `SessionManager::ExpireAllPairingsForFabric` call, I would think.

Reply:
Noticed this API, but didn't figure out a way to have the reference to SessionManager instance in FailSafeCleanup.

It looks the API is broken if caseSessionManager->ReleaseSessionsForFabric just do nothing, we should fix this API, right?